File: app/telemetry.go
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

package app

import "github.com/mattermost/mattermost-server/v6/services/telemetry"

func (s *Server) GetTelemetryService() *telemetry.TelemetryService {
	return s.telemetryService
}

File: webapp/platform/client/src/telemetry.ts
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export interface TelemetryHandler {
    trackEvent: (userId: string, userRoles: string, category: string, event: string, props?: Record<string, unknown>) => void;
    trackFeatureEvent: (userId: string, userRoles: string, featureName: string, event: string, props?: Record<string, unknown>) => void;
    pageVisited: (userId: string, userRoles: string, category: string, name: string) => void;
}
